### PR TITLE
[GnuPG] Add builder

### PR DESCRIPTION
**Pull request by Wimmerer**
```
[18:13:59] /workspace/destdir/lib/libgnutls.so: undefined reference to `nettle_des_decrypt@NETTLE_6'
[18:13:59] /workspace/destdir/lib/libgnutls.so: undefined reference to `nettle_dsa_params_clear@HOGWEED_4'
[18:13:59] /workspace/destdir/lib/libgnutls.so: undefined reference to `nettle_ecc_size_a@HOGWEED_4'
[18:13:59] /workspace/destdir/lib/libgnutls.so: undefined reference to `nettle_ccm_encrypt_message@NETTLE_6'
[18:13:59] /workspace/destdir/lib/libgnutls.so: undefined reference to `nettle_dsa_generate_params@HOGWEED_4'
[18:13:59] /workspace/destdir/lib/libgnutls.so: undefined reference to `nettle_rsa_pkcs1_verify@HOGWEED_4'
[18:13:59] /workspace/destdir/lib/libgnutls.so: undefined reference to `nettle_sha256_digest@NETTLE_6'
[18:13:59] /workspace/destdir/lib/libgnutls.so: undefined reference to `nettle_hmac_sha1_digest@NETTLE_6'
[18:13:59] collect2: error: ld returned 1 exit status
[18:13:59] make[2]: *** [Makefile:913: dirmngr] Error 1
[18:13:59] make[2]: *** Waiting for unfinished jobs....
[18:13:59] mv -f .deps/t_http_basic-dns.Tpo .deps/t_http_basic-dns.Po
[18:14:00] mv -f .deps/t_ldap_parse_uri-dns.Tpo .deps/t_ldap_parse_uri-dns.Po
[18:14:00] make[2]: Leaving directory '/workspace/srcdir/gnupg-2.2.27/dirmngr'
[18:14:00] make[1]: *** [Makefile:619: all-recursive] Error 1
[18:14:00] make[1]: Leaving directory '/workspace/srcdir/gnupg-2.2.27'
[18:14:00] make: *** [Makefile:539: all] Error 2
[18:14:00]  ---> make -j${nproc}
[18:14:00] Previous command exited with 2
[18:14:00] Child Process exited, exit code 2
```

### CODE REVIEW

File: G/GnuPG/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    Dependency("Zlib_jll"),
+    Dependency("Libassuan_jll"),
+    Dependency("OpenLDAPClient_jll"),
+    Dependency("Bzip2_jll"),
+    Dependency("SQLite_jll"),
+    Dependency("libusb_jll")
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies)
```
Comment (giordano) [inline G/GnuPG/build_tarballs.jl:50]:
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies)

```

**Comment by giordano** [comment]
The problem mentioned above was that GnuTLS was built with an older version of Nettle.  We need to be consistent, so I fixed the compat of Nettle in our dependencies to the same version used to build GnuTLS

**Comment by Wimmerer** [comment]
Windows is due to [this](https://github.com/JuliaPackaging/Yggdrasil/blob/master/G/GnuTLS/build_tarballs.jl#L35)

**Comment by Wimmerer** [comment]
There's no libassuan.dll in the lib directory for windows platforms. Not sure why, it seems like it's generated by libassuan_jll...

**Comment by giordano** [comment]
I don't see it in https://github.com/JuliaBinaryWrappers/Libassuan_jll.jl/releases/download/Libassuan-v2.5.5%2B0/Libassuan.v2.5.5.x86_64-w64-mingw32.tar.gz

File: G/GnuPG/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    Dependency("Libgcrypt_jll"),
+    Dependency("Libgpg_error_jll"),
+    Dependency("nPth_jll"),
+    Dependency("Zlib_jll"),
+    Dependency("Libassuan_jll"),
+    Dependency("OpenLDAPClient_jll"),
+    Dependency("Bzip2_jll"),
+    Dependency("SQLite_jll"),
+    Dependency("libusb_jll"),
+    Dependency("Nettle_jll", v"3.4.1", compat="~3.4.1"),
```
Comment (giordano) [inline G/GnuPG/build_tarballs.jl:77]: I need to double check this, but I think a better solution is to fix this in the registry to make gnutls require a specific version of nettle, especially if gnupg doesn't directly require nettle

File: G/GnuPG/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/gnupg-*/
+if [[ "${target}" != ${MACHTYPE} ]]; then
+    # Delete `gpg-error-config` of the host to prevent it from being picked up
+    # when configuring the package
+    rm "${host_bindir}/gpg-error-config"
+fi
```
Comment (giordano) [inline G/GnuPG/build_tarballs.jl:22]:
```suggestion
    export LDFLAGS="-L${libdir}"
fi
```

**Comment by Wimmerer** [comment]
I killed it by accident

**Pull request closed without merging**